gridstatus/lmp_config.py

```diff
@@ -0,0 +1,154 @@
+import datetime
+import functools
+import inspect
+
+import pandas as pd
+
+from gridstatus.base import ISOBase, Markets, NotSupported
+
+
+class lmp_config:
+
+    configs = {}
+
+    def __init__(self, supports, tz=None):
+        self.supports = supports
+        self.tz = tz
+
+    def __call__(self, func):
+
+        lmp_config.configs[func.__qualname__] = self.supports
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            bound_args = self._get_bound_args(func, args, kwargs)
+            if len(args) > 0 and isinstance(args[0], ISOBase):
+                bound_args = self._verify_bound_args(bound_args)
+                return self._class_method_wrapper(func, bound_args)
+            else:
+                raise ValueError("Must be class method on ISOBase")
```

Comment:
any way to do this validation check outside of `wrapper` so that it get checked before the method is actually called?

im searching and i can't find an easy way...

Reply:
i think we can leave it as-is for now. 

perhaps just leave a comment in the code  for posterity that this check has to happen at function runtime

Reply:
Done